### PR TITLE
chore: add tests for the missing functionality

### PR DESCRIPTION
**Pull request by attiyaIshaque**
### Description

I've added unit tests to the forgot password feature to improve coverage. As a result, the coverage has increased. Additionally, I've corrected some variable names due to incorrect spellings.

#### JIRA

[VAN-2004](https://2u-internal.atlassian.net/browse/VAN-2004)

#### How Has This Been Tested?

Mention any tests you've added or updated to ensure the changes work as expected.

#### Screenshots/sandbox (optional):

Include a link to the sandbox for design changes or screenshot for before and after. **Remove this section if it's not applicable.**

|Before|After|
|-------|-----|
| Screenshot 2024-07-11 at 7 07 04 PM| Screenshot 2024-07-11 at 7 01 31 PM |
| Screenshot 2024-07-11 at 7 08 54 PM| Screenshot 2024-07-11 at 7 01 39 PM|
|Screenshot 2024-07-11 at 7 09 43 PM | Screenshot 2024-07-11 at 7 01 47 PM|
|Screenshot 2024-07-11 at 7 10 21 PM | Screenshot 2024-07-11 at 7 01 53 PM|
|Screenshot 2024-07-11 at 7 10 59 PM |Screenshot 2024-07-11 at 7 02 04 PM|

#### Merge Checklist

* [ ] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Sandbox, if applicable.
* [ ] Is there adequate test coverage for your changes?

### CODE REVIEW

File: src/forms/reset-password-popup/reset-password/messages.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ const messages = defineMessages({
     defaultMessage: 'Please check your responses and try again.',
     description: 'Error message for reset password page',
   },
-  resetPassowrdSuccess: {
+  resetPasswordSuccess: {
```
Comment (ahtesham-quraish) [inline src/forms/reset-password-popup/reset-password/messages.js:10]: did we fix it in the component as well ?

Comment (attiyaIshaque) [reply src/forms/reset-password-popup/reset-password/messages.js:10]: yes

File: src/forms/reset-password-popup/reset-password/tests/index.test.jsx [hunk 2]
```diff
@@ -252,4 +291,42 @@ describe('ResetPasswordPage', () => {
...
+        status: TOKEN_STATE.VALID,
+      },
+    });
+
+    render(reduxWrapper(<IntlResetPasswordPage />));
+
+    const submitButton = screen.getByRole('button', { name: /Reset password/i, id: 'submit-new-password' });
+    fireEvent.mouseDown(submitButton);
+
+    expect(submitButton).toBeTruthy();
```
Comment (mubbsharanwar) [inline src/forms/reset-password-popup/reset-password/tests/index.test.jsx:330]: The ask of unit test is to check that mouseDown event is fired on submit button click but we are not checking that event is fired or not

**Comment by ahtesham-quraish** [comment]
@attiyaIshaque please rebase the branch before merging

**Review by ahtesham-quraish — APPROVED** [review]

File: src/forms/reset-password-popup/reset-password/tests/ResetPasswordSuccess.test.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+jest.mock('@edx/frontend-platform/i18n', () => ({
+  useIntl: () => ({
+    formatMessage: jest.fn((message) => message.defaultMessage),
+  }),
+  // eslint-disable-next-line no-shadow
+  defineMessages: (messages) => messages,
+}));
+
+describe('ResetPasswordSuccess component', () => {
+  it('renders success alert with correct message', () => {
```
Comment (mubbsharanwar) [inline src/forms/reset-password-popup/reset-password/tests/ResetPasswordSuccess.test.jsx:18]: Could you explain this unit test?

Comment (attiyaIshaque) [reply src/forms/reset-password-popup/reset-password/tests/ResetPasswordSuccess.test.jsx:18]: It verifies that the ResetPasswordSuccess component renders a success alert with the correct message

File: src/forms/reset-password-popup/reset-password/tests/index.test.jsx [hunk 4]
```diff
@@ -163,7 +163,38 @@ describe('ResetPasswordPage', () => {
...
+      ...initialState,
+      resetPassword: {
+        status: PASSWORD_VALIDATION_ERROR,
+      },
+    });
+    render(reduxWrapper(<IntlResetPasswordPage />));
+    const newPasswordInput = screen.getByLabelText('New password');
+    fireEvent.blur(newPasswordInput);
+
+    const backendValidationError = screen.getByText('Password is a required field');
```
[the remark this answers is not in the document]
Comment (attiyaIshaque) [reply src/forms/reset-password-popup/reset-password/tests/index.test.jsx:193]: Yes. But i am check backend validation here.

File: src/forms/reset-password-popup/reset-password/tests/index.test.jsx (hunk 2 above)
Comment (attiyaIshaque) [reply src/forms/reset-password-popup/reset-password/tests/index.test.jsx]: Thank you, updated tests

File: src/forms/reset-password-popup/reset-password/tests/index.test.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (mubbsharanwar) [reply src/forms/reset-password-popup/reset-password/tests/index.test.jsx:193]: But this error message is generating from frontend validations, remove the line no. 190 and 191 then backend validations code will be execute.

File: src/forms/reset-password-popup/reset-password/tests/ResetPasswordSuccess.test.jsx (hunk 3 above)
Comment (mubbsharanwar) [reply src/forms/reset-password-popup/reset-password/tests/ResetPasswordSuccess.test.jsx:18]: Where we are checking success alert exists.
It should be like `expect(getByText(successMessage)).toBeTruthy();`

Comment (attiyaIshaque) [reply src/forms/reset-password-popup/reset-password/tests/ResetPasswordSuccess.test.jsx:17]: > Where we are checking success alert exists. It should be like `expect(getByText(successMessage)).toBeTruthy();`

Done

**Review by mubbsharanwar — APPROVED** [review]

**Pull request merged**